pennylane/_qutrit_device.py

```diff
@@ -0,0 +1,383 @@
+
+        # count the basis state occurrences, and construct the probability vector
+        if bin_size is not None:
+            bins = len(samples) // bin_size
+
+            indices = indices.reshape((bins, -1))
+            prob = np.zeros([3 ** len(device_wires), bins], dtype=np.float64)
+
+            for b, idx in enumerate(indices):
+                basis_states, counts = np.unique(idx, return_counts=True)
+                prob[basis_states, b] = counts / bin_size
+
+        else:
+            basis_states, counts = np.unique(indices, return_counts=True)
+            prob = np.zeros([3 ** len(device_wires)], dtype=np.float64)
+            prob[basis_states] = counts / len(samples)
+
+        return self._asarray(prob, dtype=self.R_DTYPE)
+
+    def marginal_prob(self, prob, wires=None):
+        r"""Return the marginal probability of the computational basis
+        states by summing the probabiliites on the non-specified wires.
+
+        If no wires are specified, then all the basis states representable by
+        the device are considered and no marginalization takes place.
+
+        .. note::
+
+            If the provided wires are not in the order as they appear on the device,
+            the returned marginal probabilities take this permutation into account.
+
+            For example, if the addressable wires on this device are ``Wires([0, 1, 2])`` and
+            this function gets passed ``wires=[2, 0]``, then the returned marginal
+            probability vector will take this 'reversal' of the two wires
+            into account:
+
+            .. math::
+
+                \mathbb{P}^{(2, 0)}
+                            = \left[
+                               |00\rangle, |10\rangle, |01\rangle, |11\rangle
+                              \right]
+
+        Args:
+            prob: The probabilities to return the marginal probabilities
+                for
+            wires (Iterable[Number, str], Number, str, Wires): wires to return
+                marginal probabilities for. Wires not provided
+                are traced out of the system.
+
+        Returns:
+            array[float]: array of the resulting marginal probabilities.
+        """
+
+        if wires is None:
+            # no need to marginalize
+            return prob
+
+        wires = Wires(wires)
+        # determine which subsystems are to be summed over
+        inactive_wires = Wires.unique_wires([self.wires, wires])
+
+        # translate to wire labels used by device
+        device_wires = self.map_wires(wires)
+        inactive_device_wires = self.map_wires(inactive_wires)
+
+        # reshape the probability so that each axis corresponds to a wire
+        prob = self._reshape(prob, [3] * self.num_wires)
+
+        # sum over all inactive wires
+        prob = self._flatten(self._reduce_sum(prob, inactive_device_wires.labels))
+        # TODO: Add case for when inactive_device_wires is not an instance of Wires once
```

Comment:
No, this is not the case. That TODO is there because `inactive_device_wires` will not be wrapped inside a `Wires` object once `default.qutrit` is added and is being used. However, `QutritDevice` does wrap `inactive_device_wires` in a `Wires` object. Therefore, there will be a check here in the future to ensure that the prob array is created correctly if the device being used is a `DefaultQutrit` object.